install.sh

```diff
@@ -94,133 +103,128 @@ esac
 
 # functions to maintain packages
 function download_check () {
-    cd $CREW_BREW_DIR
+    cd "${CREW_BREW_DIR}"
 
     #download
-    echo "Downloading $1..."
-    curl --progress-bar -C - -# -L --ssl $2 -o "$3"
+    echo "Downloading ${1}..."
+    curl --progress-bar -C - -L --ssl "${2}" -o "${3}"
 
     #verify
-    echo "Verifying $1..."
-    echo $4 $3 | sha256sum -c -
-    case $? in
+    echo "Verifying ${1}..."
+    echo "${4}" "${3}" | sha256sum -c -
+    case "${?}" in
     0) ;;
     *)
-      echo "Verification failed, something may be wrong with the $1 download."
+      echo "Verification failed, something may be wrong with the download."
       exit 1;;
     esac
 }
 
 function extract_install () {
-    cd $CREW_BREW_DIR
+    cd "${CREW_BREW_DIR}/dest"
 
     #extract and install
-    echo "Extracting $1 (this may take a while)..."
-    rm -rf ./usr ./home
-    tar -xf $2
-    echo "Installing $1 (this may take a while)..."
-    if [ -d "${CREW_BREW_DIR}/home" ]; then
-      tar cf - ./usr/* ./home/* | (cd /; tar xp --keep-directory-symlink -f -)
-    else
-      tar cf - ./usr/* | (cd /; tar xp --keep-directory-symlink -f -)
-    fi
-    mv ./dlist $CREW_CONFIG_PATH/meta/$1.directorylist
-    mv ./filelist $CREW_CONFIG_PATH/meta/$1.filelist
+    echo "Extracting ${1} (this may take a while)..."
+    for dir in ./*/; do
+      rm -rf "${dir}"
+    done
+    for file in ./*list; do
+      rm -f "${file}"
+    done
+    tar xpf ../"${2}"
+    echo "Installing ${1} (this may take a while)..."
+    tar cpf - ./*/* | (cd /; tar xp --keep-directory-symlink -f -)
+    mv ./dlist "${CREW_CONFIG_PATH}/meta/${1}.directorylist"
+    mv ./filelist "${CREW_CONFIG_PATH}/meta/${1}.filelist"
 }
 
 function update_device_json () {
-  cd $CREW_CONFIG_PATH
+  cd "${CREW_CONFIG_PATH}"
 
-  if grep '"name": "'$1'"' device.json > /dev/null; then
-    echo "Updating version number of existing $1 information in device.json..."
-    sed -i device.json -e '/"name": "'$1'"/N;//s/"version": ".*"/"version": "'$2'"/'
+  if grep '"name": "'"${1}"'"' device.json > /dev/null; then
```

Comment:
Quoting is not necessary here and only makes things more confusing.

Reply:
I don't like the idea of unquoted variables. All it does is add a pair of `"`.

Reply:
Uh, dude, the variables are already quoted.  Why do I get the idea you did a global search and replace blindly?

Reply:
The quotes are cancelled out. Currently, `${1}` was unquoted. Quote stacking is a tricky topic, but I have made sure this specific string works.

Reply:
`if grep '"name": "'${1}'"' device.json > /dev/null; then`

Reply:
Yes. The single quotes surrounded the double quotes and made them literal. The `${1}` was unquoted.

Reply:
Nothing is made literal here.  Look at the line carefully.

Reply:
The double quotes are taken literally.
I have looked at it carefully.
If you'd like to keep arguing, go ahead but I know it's correct because I've tested it.

Reply:
I don't think you are listening.

Reply:
I am, but I need to be listened to.
Please see [Quoting](https://www.gnu.org/software/bash/manual/html_node/Quoting.html).
Maybe then it will make more sense :).